validator_client/validator_services/src/sync.rs

```diff
@@ -607,187 +606,81 @@ pub async fn fill_in_aggregation_proofs<S: ValidatorStore, T: SlotClock + 'stati
-                                    %subnet_id,
-                                    "Validator is sync aggregator"
-                                );
-                                Some(((proof_slot, *subnet_id), proof))
-                            }
-                            Ok(false) => None,
-                            Err(e) => {
-                                warn!(
-                                    pubkey = ?duty.pubkey,
-                                    slot = %proof_slot,
-                                    error = ?e,
-                                    "Error determining is_aggregator"
-                                );
-                                None
-                            }
-                        },
-
-                        None => None,
-                    }
+                    result.map(|proof| (duty.validator_index, proof_slot, subnet_id, proof))
                 });
-
-                // Execute all the futures in parallel, collecting any successful results.
-                let proofs = join_all(futures)
-                    .await
-                    .into_iter()
-                    .flatten()
-                    .collect::<Vec<_>>();
-
-                validator_proofs.push((duty.validator_index, proofs));
             }
+        }
 
-            // Add to global storage (we add regularly so the proofs can be used ASAP).
+        while let Some(result) = futures_unordered.next().await {
+            let Some((validator_index, proof_slot, subnet_id, proof)) = result else {
+                continue;
+            };
             let sync_map = duties_service.sync_duties.committees.read();
             let Some(committee_duties) = sync_map.get(&sync_committee_period) else {
-                debug!(period = sync_committee_period, "Missing sync duties");
+                debug!("period" = sync_committee_period, "Missing sync duties");
                 continue;
             };
+
             let validators = committee_duties.validators.read();
-            let num_validators_updated = validator_proofs.len();
 
-            for (validator_index, proofs) in validator_proofs {
-                if let Some(Some(duty)) = validators.get(&validator_index) {
-                    duty.aggregation_duties.proofs.write().extend(proofs);
-                } else {
-                    debug!(
+            // Check if the validator is an aggregator
+            match proof.is_aggregator::<S::E>() {
+                Ok(true) => {
+                    if let Some(Some(duty)) = validators.get(&validator_index) {
+                        debug!(
+                            validator_index,
+                            "slot" = %proof_slot,
+                            "subcommittee_index" = *subnet_id,
+                            // log full selection proof for debugging
+                            "full selection proof" = ?proof,
+                            "Validator is sync aggregator"
```

Comment:
In the interest of combining both normal and distributed cases, the "full selection proof" will make it confusing, I suggest we just stick to selection_proof, and keep the comment to indicate it is a full selection proof

Also we can remove the double quotes in the fields, I am not sure why I put them there in the past

```suggestion
                            validator_index,
                            slot = %proof_slot,
                            subcommittee_index = *subnet_id,
                            // In distributed mode, this is the full selection proof
                            selection_proof = ?proof,
                            "Validator is sync aggregator"
```